sxs/formats/lvc/metadata.py

```diff
@@ -0,0 +1,291 @@
+    SXS ID numbers and whose values are dictionaries including masses
+    and spins) and a dictionary catalog_resolutions (whose keys are SXS
+    ID numbers and whose values are dictionaries containing lists of
+    integers indicating what resolutions are available in the catalog
+    for that SXS ID number), return an LVC-format filename that is a
+    comparable simulation with multiple resolutions available.
+
+    """
+    mass1 = catalog[sxs_id]['initial_mass1']
+    mass2 = catalog[sxs_id]['initial_mass2']
+    spin1 = catalog[sxs_id]['initial_dimensionless_spin1']
+    spin2 = catalog[sxs_id]['initial_dimensionless_spin2']
+    mass_ratio = mass1 / mass2
+    spin1_magnitude = np.linalg.norm(spin1)
+    spin2_magnitude = np.linalg.norm(spin2)
+
+    # Select SXS ID numbers with multiple resolutions available
+    has_multiple_resolutions = []
+    for key in catalog_resolutions:
+        if len(catalog_resolutions[key]) > 1:
+            has_multiple_resolutions.append(key)
+
+    # Select SXS ID numbers with multiple resolutions and the same
+    # initial data type and spec revision
+    same_id_and_spec_revision = []
+    for key in has_multiple_resolutions:
+        if (catalog[key]['spec_revisions'] == catalog[sxs_id]['spec_revisions']
+            and catalog[key]['initial_data_type'] == catalog[sxs_id]['initial_data_type']):
+            same_id_and_spec_revision.append(key)
+
+    if len(same_id_and_spec_revision) > 0:
+        has_multiple_resolutions = same_id_and_spec_revision
+
+    # Select an SXS ID number with multiple resolutions that is "similar" to
+    # sxs_id. First, try to find a case with the same initial data type and
+    # SpEC revision. If that is possible, then choose from among the ones with
+    # the closest mass ratio and spin magnitudes.  Otherwise, just select a
+    # simulation with the closest mass ratios and spin magnitudes.
+    #
+    # Note: as in the previous script, here I use initial masses and spins, not
+    # relaxed masses and spins.
+    mass_spin_diff_best = 1.e100
+    key_best = ""
+    for key in has_multiple_resolutions:
+        current_mass1 = catalog[key]['initial_mass1']
+        current_mass2 = catalog[key]['initial_mass2']
+        current_spin1 = catalog[key]['initial_dimensionless_spin1']
+        current_spin2 = catalog[key]['initial_dimensionless_spin2']
+        current_mass_ratio = current_mass1 / current_mass2
+        current_spin1_magnitude = np.linalg.norm(current_spin1)
+        current_spin2_magnitude = np.linalg.norm(current_spin2)
+        mass_spin_diff = (
+                np.abs(mass_ratio - current_mass_ratio)
+                + np.abs(spin1_magnitude - current_spin1_magnitude)
+                + np.abs(spin2_magnitude - current_spin2_magnitude)
+        )
+        if mass_spin_diff < mass_spin_diff_best:
+            mass_spin_diff_best = mass_spin_diff
+            key_best = key
+
+    resolution_best = np.max(catalog_resolutions[key])
```

Comment:
@geoffrey4444 I'm almost ready to test this, but I found one issue that I think you need to take a look at.

Presumably `key` in this line is supposed to be `key_best`.  Unfortunately, the code as is (which is presumably what you've used in the past) will actually run without notifying you of any issues, but the value of `key` in this line is just the last value from the loop — which I guess may or may not actually be the best.  So it seems likely that there are some incorrect choices of resolution in what you've produced with this.

Also, it gives me the heebie-jeebies to define `key_best = ""`; I'd prefer to define it as `key_best = has_multiple_resolutions[0]`.

Do these two changes make sense to you?